### PR TITLE
Issue-16: Bug Fixes and dependency updates

### PR DESCRIPTION
**Pull request by mogmarsh**
## Description

This PR addresses issue #16. We no longer need the Omeda integration so this this branch only aims now to update npm dependencies and it also includes several other bug fixes we will need in the plugin regardless.

## Related Issue

https://github.com/alleyinteractive/wp-newsletter-builder/issues/16

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/class-rest-api-endpoints.php [hunk 1]
```diff
@@ -25,15 +28,16 @@ public function __construct() {
...
 		register_rest_route(
 			'wp-newsletter-builder/v1',
 			'/lists/',
 			[
 				'methods'             => 'GET',
 				'callback'            => [ $this, 'get_lists' ],
 				'permission_callback' => function () {
-					return current_user_can( 'edit_posts' );
+					return true;
+					// return current_user_can( 'edit_posts' ); TODO.
```
Comment (stevenslack) [inline src/class-rest-api-endpoints.php:40]: Remove this TODO. This should exist in an issue.

File: single-nb_newsletter.php [hunk 2]
```diff
@@ -2,7 +2,7 @@
 /**
  * The template for displaying Newsletters.
  *
- * @package WP_Newsletter_Builder
+ * @package wp-newsletter-builder
```
Comment (stevenslack) [inline single-nb_newsletter.php:5]: What is the reason for this change? Is it to use the namespace of the plugin name rather than the PHP namespace?

File: src/email-providers/class-campaign-monitor.php [hunk 3]
```diff
@@ -106,6 +106,7 @@ public function get_lists() {
 	 * @return array The response from the API.
 	 */
 	public function create_campaign( $newsletter_id, $list_ids, $campaign_id = null ) {
+		// TODO: Move non-email provider code to the core plugin.
```
Comment (stevenslack) [inline src/email-providers/class-campaign-monitor.php:109]: Remove TODOs

**Review by stevenslack — COMMENTED** [review]
The Todos should be removed and a couple of other questions

File: single-nb_newsletter.php (hunk 2 above)
Comment (attackant) [reply single-nb_newsletter.php:5]: Previously, both versions were in the plugin (from Greg's work) most of them were using wp-newsletter-builder -- so this was to make the usages consistent. I am not sure which is prefferable.

File: src/class-rest-api-endpoints.php (hunk 1 above)
Comment (attackant) [reply src/class-rest-api-endpoints.php:40]: Here is my situation. This is a work in progress. We are waiting on client instructions before continuing development on the email integration. I would like to get these changes merged into develop so that Greg's work is using all of the same dependencies and some bug fixes here are included. We will be making new issues to handle these, but are waiting on client instructions before doing so, however I would prefer this doesn't stop all progress. This plugin is not being used on any sites yet. While this is not the standard practice, I think it makes sense to get this merged ahead of completing this work or removing these code comments. Open to your thoughts though, I am not fixated on this. Maybe I need to cherry pick commits or keep working on this, but it will need to be ticketed as PD if we do that, if we wait until we get client instructions, we can easily make it a part of our billable work.

**Review by stevenslack — APPROVED** [review]
